cluster/src/main/java/org/apache/iotdb/cluster/query/ClusterPlanExecutor.java

```diff
@@ -121,38 +124,41 @@ public QueryDataSet processQuery(PhysicalPlan queryPlan, QueryContext context)
         synchronized (response) {
           try {
             if (client != null) {
-              client.getNodeList(null, schemaPattern, level, handler);
+              client.getNodeList(node, schemaPattern, level, handler);
               response.wait(connectionTimeoutInMS);
             }
           } catch (TException e) {
             logger.error("Error occurs when getting node lists in node {}.", node, e);
```

Comment:
It will be more robust if, for each group, you will try querying each node in the group until one result is returned. So that even if one node is down, you can still get the result from its replicas.

Reply:
Thanks for your suggestion.